contracts/lib/FulfillmentApplier.sol

```diff
@@ -225,262 +231,183 @@ contract FulfillmentApplier is FulfillmentApplicationErrors {
+                        mload(add(offerItemPtr, Common_token_offset))
+                    )
 
-                // Zero out amount on original item to indicate it is credited.
-                mstore(amountPtr, 0)
+                    // Set the identifier on the received item.
+                    mstore(
+                        add(receivedItem, Common_identifier_offset),
+                        mload(add(offerItemPtr, Common_identifier_offset))
+                    )
+
+                    // Set the offerer on returned execution using order pointer.
+                    mstore(
+                        add(execution, Execution_offerer_offset),
+                        mload(paramsPtr)
+                    )
+
+                    // Set returned execution conduitKey via order pointer offset.
+                    mstore(
+                        add(execution, Execution_conduit_offset),
+                        mload(add(paramsPtr, OrderParameters_conduit_offset))
+                    )
+
+                    // Calculate the hash of (itemType, token, identifier).
+                    dataHash := keccak256(
+                        receivedItem,
+                        ReceivedItem_CommonParams_size
+                    )
 
-                // Ensure the indicated item matches original item.
-                if iszero(
-                    and(
-                        and(
-                          // The offerer must match on both items.
-                          eq(
-                              mload(paramsPtr),
-                              mload(
-                                  add(execution, Execution_offerer_offset)
-                              )
-                          ),
-                          // The conduit key must match on both items.
-                          eq(
-                              mload(
-                                  add(
-                                      paramsPtr,
-                                      OrderParameters_conduit_offset
-                                  )
-                              ),
-                              mload(
-                                  add(
-                                      execution,
-                                      Execution_conduit_offset
-                                  )
-                              )
-                          )
+                    // If component index > 0, swap component pointer with pointer
+                    // to first component.
+                    let firstFulfillmentHeadPtr := add(offerComponents, OneWord)
+                    if xor(firstFulfillmentHeadPtr, fulfillmentHeadPtr) {
+                      let firstFulfillmentPtr := mload(firstFulfillmentHeadPtr)
+                      let fulfillmentPtr := mload(fulfillmentHeadPtr)
+                      mstore(firstFulfillmentHeadPtr, fulfillmentPtr)
+                      mstore(fulfillmentHeadPtr, firstFulfillmentPtr)
```

Comment:
This makes sure that when there is a leftover unmatched aggregated offer item amounts, this amount will be overwritten to (now) the first used offer element that is pointed to by the first offer fulfillment element.

The PR does make a similar change on the consideration aggregation side. So for example, when `Seaport` tries to [store a leftover amount](https://github.com/ProjectOpenSea/seaport/blob/c7c1645e149b4e6840cb7e4f92d687eb5ba8fbcc/contracts/lib/FulfillmentApplier.sol#L105-L117) for consideration if the first element of the consideration fulfillment points to a consideration index that is OOR the transaction would fail (even though, we skipped this item in the aggregation subroutine).